fvm/evm/emulator/emulator.go

```diff
@@ -377,14 +377,15 @@ func (proc *procedure) withdrawFrom(
 	if !isValid {
 		return types.NewInvalidResult(
 			call.Transaction(),
-			types.ErrInvalidBalance), nil
+			types.ErrInvalidBalance,
+		), nil
 	}
 
-	// check balance is not prone to rounding error
-	if types.BalanceConversionToUFix64ProneToRoundingError(call.Value) {
+	if !types.AttoFlowBalanceValidForFlowVault(call.Value) {
```

Comment:
Is this change needed, since we already changed the behavior in  `newInternalEVMTypeWithdrawFunction` so that we are guaranteed that the value will be rounded to UFix precision?
Is there another path we want to cover with this change?

Reply:
The `newInternalEVMTypeWithdrawFunction` affects only the `InternalEVM.withdraw` function(https://github.com/onflow/flow-go/blob/master/fvm/evm/stdlib/contract.cdc#L454C26-L457), which is currently being used by `CadenceOwnedAccount.withdraw`. However, the `BlockView.DirectCall` might possibly get its input from somewhere else, e.g. the flow-go side or whatever. And since the same check was performed here, regarding the rounding error, I am inclined to leave the same check in as a redundancy, just in case.